src/System.Text.JsonLab.Serialization/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -0,0 +1,273 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+
+namespace System.Text.Json.Serialization
+{
+    internal class JsonClassInfo
+    {
+        // The length of the property name embedded in the key (in bytes).
+        private const int PropertyNameKeyLength = 6;
```

Comment:
What's the rationale behind choosing 6 for the key?

Reply:
The underlying propertyname->info "hashtable" is a List<ulong> and 6 bytes (the first 6 characters of the property name) are reserved for the key and 2 bytes for the length. This allows a high-performance iteration since just ulongs are compared for the key. Once a hit is made based on key, the remaining bytes are searched if >6. Once a match is found then a secondary array is accessed with the same index that returns the cached "info" structure for the property.

Reply:
Ah gotcha. Makes sense. Thanks for the explanation.